apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -97,3 +123,30 @@ func (api *CredentialValidatorAPI) ModelCredential() (params.ModelCredential, er
 		Valid:           c.Valid,
 	}, nil
 }
+
+// Mask out new methods from the old API versions. The API reflection
+// code in rpc/rpcreflect/type.go:newMethod skips 2-argument methods,
+// so this removes the method as far as the RPC machinery is concerned.
+//
+// WatchModelCredential did not exist prior to v2.
+func (*CredentialValidatorAPIV1) WatchModelCredential(_, _ struct{}) {}
+
+// WatchModelCredential returns a NotifyWatcher that watches what cloud credential a model uses.
```

Comment:
Maybe this would be clearer as "returns a NotifyWatcher that signals when the model's cloud credential changes."?

Reply:
Thank you for the tip :) I looked at common.MultiNotifyWatcher as per your suggestion... I am not too sure that it will actually be of much benefit to me here. We can be in situation where we do not have a credential to watch but we do want to watch a model-credential reference changing from nil/empty to something concrete. It feels like an easier job done with 2 watchers.

I'll keep it as-is for now. We can circle back if needed.